x-pack/heartbeat/monitors/browser/source/zipurl_test.go

```diff
@@ -5,69 +5,132 @@
 package source
 
 import (
-	"context"
 	"fmt"
 	"net/http"
+	"net/http/httptest"
 	"os"
 	"path"
 	"path/filepath"
 	"runtime"
 	"testing"
 
 	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
 
+	"github.com/elastic/beats/v7/libbeat/common"
 	"github.com/elastic/beats/v7/x-pack/heartbeat/monitors/browser/source/fixtures"
 )
 
-func TestZipUrlFetchNoAuth(t *testing.T) {
-	address, teardown := setupTests()
-	defer teardown()
-
-	zus := &ZipURLSource{
-		URL:     fmt.Sprintf("http://%s/fixtures/todos.zip", address),
-		Folder:  "/",
-		Retries: 3,
+func TestSimpleCases(t *testing.T) {
+	type testCase struct {
+		name         string
+		cfg          common.MapStr
+		tlsServer    bool
+		wantFetchErr bool
+	}
+	testCases := []testCase{
+		{
+			"basics",
+			common.MapStr{
+				"folder":  "/",
+				"retries": 3,
+			},
+			false,
+			false,
+		},
+		{
+			"targetdir",
+			common.MapStr{
+				"folder":           "/",
+				"retries":          3,
+				"target_directory": "/tmp/synthetics/blah",
+			},
+			false,
+			false,
+		},
+		{
+			"auth success",
+			common.MapStr{
+				"folder":   "/",
+				"retries":  3,
+				"username": "testuser",
+				"password": "testpass",
+			},
+			false,
+			false,
+		},
+		{
+			"auth failure",
+			common.MapStr{
+				"folder":   "/",
+				"retries":  3,
+				"username": "testuser",
+				"password": "badpass",
+			},
+			false,
+			true,
+		},
+		{
+			"ssl ignore cert errors",
+			common.MapStr{
+				"folder":  "/",
+				"retries": 3,
+				"timeout": 123,
```

Comment:
wouldn't it be flaky at any point in the future?

Reply:
Good point, will remove.